fsspec/spec.py

```diff
@@ -1567,6 +1567,79 @@ def modified(self, path):
         """Return the modified timestamp of a file as a datetime.datetime"""
         raise NotImplementedError
 
+    def tree(
+            self,
+            path: str = '/',
+            recursion_limit: int = 2,
+            max_display: int = 25,
+            display_size: bool = False,
+            prefix: str = "",
+            is_last: bool = True,
+            first: bool = True
```

Comment:
I'll work on that.

Reply:
Could you clarify what you mean? If I understand correctly, the `tree` method would output a `str` directly, but it would be displayed as a 'flattened' string, showing the `\n` characters and losing the tree's visual structure. May I allow the user to choose between printing each line on the fly and returning the complete tree as a string?

Reply:
If you return a string, the user may with to print that:
```
print(fs.tree(...))
```
but could equally push it to some file, console/log or other output. For example, the string output would make a lot of sense in a text UI component.